controllers/webauthn.go

```diff
@@ -0,0 +1,133 @@
+package controllers
+
+import (
+	"bytes"
+	"io/ioutil"
+
+	"github.com/astaxie/beego/logs"
+	"github.com/casdoor/casdoor/object"
+	"github.com/casdoor/casdoor/util"
+	"github.com/duo-labs/webauthn/protocol"
+	"github.com/duo-labs/webauthn/webauthn"
+)
+
+// @Title WebAuthnRegisterBegin
+// @Tag User API
+// @Description WebAuthn Registration Flow 1st stage
+// @Success 200 {object} protocol.CredentialCreation The CredentialCreationOptions object
+// @router /webauthn/register/begin [get]
+func (c *ApiController) WebAuthnRegisterBegin() {
```

Comment:
In Casdoor, we usually use:

- `signup` instead of `register`
- `signin` instead of `login`